cmd/spire-server/cli/federation/create.go

```diff
@@ -0,0 +1,138 @@
+}
+
+func (c *createCommand) AppendFlags(f *flag.FlagSet) {
+	f.StringVar(&c.path, "data", "", "Path to a file containing federation relationships in JSON format (optional). If set to '-', read the JSON from stdin.")
+	f.StringVar(&c.trustDomain, "trustDomain", "", `Name of the trust domain to federate with (e.g., example.org)`)
+	f.StringVar(&c.bundleEndpointURL, "bundleEndpointURL", "", "URL of the SPIFFE bundle endpoint that provides the trust bundle (must use the HTTPS protocol)")
+	f.StringVar(&c.bundleEndpointProfile, "bundleEndpointProfile", profileHTTPSSPIFFE, fmt.Sprintf("Endpoint profile type (either %q or %q)", profileHTTPSWeb, profileHTTPSSPIFFE))
+	f.StringVar(&c.endpointSPIFFEID, "endpointSpiffeID", "", "SPIFFE ID of the SPIFFE bundle endpoint server. Only used for 'spiffe' profile.")
+	f.StringVar(&c.bundlePath, "bundlePath", "", "Path to the bundle data (optional). Only used for 'spiffe' profile.")
+	f.StringVar(&c.bundleFormat, "bundleFormat", util.FormatPEM, fmt.Sprintf("The format of the bundle data (optional). Either %q or %q. Only used for 'spiffe' profile.", util.FormatPEM, util.FormatSPIFFE))
+}
+
+func (c *createCommand) Run(ctx context.Context, env *common_cli.Env, serverClient util.ServerClient) error {
+	federationRelationships, err := c.getRelationships()
+	if err != nil {
+		return err
+	}
+
+	client := serverClient.NewTrustDomainClient()
+
+	resp, err := client.BatchCreateFederationRelationship(ctx, &trustdomainv1.BatchCreateFederationRelationshipRequest{
+		FederationRelationships: federationRelationships,
+	})
+	if err != nil {
+		return fmt.Errorf("request failed: %w", err)
+	}
+
+	// Process results
+	var succeeded []*trustdomainv1.BatchCreateFederationRelationshipResponse_Result
+	var failed []*trustdomainv1.BatchCreateFederationRelationshipResponse_Result
+	for i, r := range resp.Results {
+		switch r.Status.Code {
+		case int32(codes.OK):
+			succeeded = append(succeeded, r)
+		default:
+			// The trust domain API does not include in the results the relationships that
+			// failed to be created, so we populate them from the request data.
+			r.FederationRelationship = federationRelationships[i]
+			failed = append(failed, r)
+		}
+	}
+
+	// Print federation relationships that succeeded to be created
+	for _, r := range succeeded {
+		env.Println()
+		printFederationRelationship(r.FederationRelationship, env.Printf)
+	}
+
+	// Print federation relationships that failed to be created
+	for _, r := range failed {
+		env.Println()
+		env.ErrPrintf("Failed to create the following federation relationship (code: %s, msg: %q):\n",
+			codes.Code(r.Status.Code),
+			r.Status.Message)
+		printFederationRelationship(r.FederationRelationship, env.ErrPrintf)
+	}
+
+	if len(failed) > 0 {
+		return errors.New("failed to create one or more federation relationships")
+	}
+
+	return nil
+}
+
+func (c *createCommand) getRelationships() ([]*types.FederationRelationship, error) {
+	if c.path != "" {
```

Comment:
Should we verify that the other flags haven't been set with path? It might be confusing if they supply some flags and they are silently ignored.